DashAI/back/models/hugging_face/vit_transformer.py

```diff
@@ -0,0 +1,164 @@
+import json
+import shutil
+
+import numpy as np
+from sklearn.exceptions import NotFittedError
+from transformers import (
+    Trainer,
+    TrainingArguments,
+    ViTFeatureExtractor,
+    ViTForImageClassification,
+)
+
+from DashAI.back.dataloaders.classes.dashai_dataset import DashAIDataset
+from DashAI.back.models.base_model import BaseModel
+from DashAI.back.models.image_classification_model import ImageClassificationModel
+
+
+class ViTTransformer(BaseModel, ImageClassificationModel):
+    """
+    Pre-trained transformer ViT allowing image classification
+    """
+
+    @classmethod
+    def get_schema(cls):
+        with open("DashAI/back/models/parameters/models_schemas/ViT.json") as f:
+            cls.SCHEMA = json.load(f)
+        return cls.SCHEMA
+
+    def __init__(self, model=None, **kwargs):
+        """
+        Initialize the transformer class by calling the pretrained model and its
+        feature extractor. Include an attribute analogous to sklearn's check_is_fitted
+        to see if it was fine-tuned.
+        """
+        self.model_name = "google/vit-base-patch16-224"
+        self.feature_extractor = ViTFeatureExtractor.from_pretrained(self.model_name)
+        self.model = (
+            model
+            if model is not None
+            else ViTForImageClassification.from_pretrained(self.model_name)
+        )
+        self.fitted = bool(model is not None)
+        self.training_args = kwargs
+
+    def get_preprocess_images(self, input_column: str, output_column: str):
+        """Preprocess images for model input
```

Comment:
Documentation linter claims if you don't put the dot at the end 🫠

```suggestion
        """Preprocess images for model input.
```